service/src/test/java/uk/gov/mca/beacons/api/beaconsearch/BeaconSpecificationSearchServiceUnitTest.java

```diff
@@ -140,6 +142,111 @@ void givenEmailAndAccountHolder_ThenShouldCallRepositoryWithSpecificationAndSort
     );
   }
 
+  @Test
+  public void shouldCallRepositoryWithSpecificationOnlyForFullExport() {
+    BeaconSearchEntity entity1 = createDummyEntity(UUID.randomUUID());
+    BeaconSearchEntity entity2 = createDummyEntity(UUID.randomUUID());
+    List<BeaconSearchEntity> expectedList = List.of(entity1, entity2);
+
+    when(
+      beaconSearchSpecificationRepository.findAll(any(Specification.class))
+    ).thenReturn(expectedList);
+
+    List<BeaconSearchEntity> actualList =
+      beaconSpecificationSearchService.findAllBeaconsForFullExport(
+        "",
+        null,
+        null,
+        null,
+        null
```

Comment:
I think you might be able to change these parameters and the test would still pass, because we are asserting the the method returns the things it is set up to return in the `when(...).thenReturn(...)` above.